lib/Disco.php

```diff
@@ -160,15 +170,26 @@ public function handleRequest()
             HTTP::redirectTrustedURL($url);
         }
 
+        $preferredIdP = $this->getRecommendedIdP();
+        $preferredIdP = array_key_exists($preferredIdP, $idpList) ? $preferredIdP : null;
+
         // IF IS SET AUTHN CONTEXT CLASS REF, REDIRECT USER TO THE IDP
-        if (isset($this->authnContextClassRef)) {
-            if ($this->authnContextClassRef !== null) {
+        if (isset($this->originalAuthnContextClassRef)) {
+            if ($this->originalAuthnContextClassRef !== null) {
+                $state = State::loadState($this->stateId, self::SAML_SP_SSO, true);
```

Comment:
I think that you can access $state directly because it is created in the constructor.
So I don't know if it is required to load state once more.